yt_dlp/extractor/youtube.py

```diff
@@ -2287,228 +2302,122 @@ def _get_video_info_params(video_id, client='TVHTML5'):
-                    compat_str) or '{}', video_id)
-                if pr:
-                    break
-            if not pr:
-                self.report_warning('Falling back to embedded-only age-gate workaround.')
-                embed_webpage = None
-                sts = self._extract_signature_timestamp(video_id, player_url, ytcfg, fatal=False)
-                if sts and not force_mobile_client and 'configs' not in player_skip:
-                    embed_webpage = self._download_webpage(
-                        'https://www.youtube.com/embed/%s?html5=1' % video_id,
-                        video_id=video_id, note='Downloading age-gated embed config')
-
-                ytcfg_age = self._extract_ytcfg(video_id, embed_webpage) or {}
-                # If we extracted the embed webpage, it'll tell us if we can view the video
-                embedded_pr = self._parse_json(
-                    try_get(ytcfg_age, lambda x: x['PLAYER_VARS']['embedded_player_response'], str) or '{}',
-                    video_id=video_id)
-                embedded_ps_reason = try_get(embedded_pr, lambda x: x['playabilityStatus']['reason'], str) or ''
-                if embedded_ps_reason not in self._AGE_GATE_REASONS:
-                    yt_client = 'WEB_EMBEDDED_PLAYER'
-                    if not sts or force_mobile_client:
-                        # Android client already has signature descrambled
-                        # See: https://github.com/TeamNewPipe/NewPipeExtractor/issues/562
-                        if not sts:
-                            self.report_warning(
-                                'Falling back to android embedded client for player API (note: some formats may be missing).')
-                        yt_client = 'ANDROID_EMBEDDED_PLAYER'
-                        ytcfg_age = {}
-
-                    ytage_headers = self._generate_api_headers(
-                        ytcfg_age, identity_token, syncid,
-                        client=yt_client, session_index=session_index)
-                    yt_age_query = {'videoId': video_id}
-                    yt_age_query.update(self._generate_player_context(sts))
-                    pr = self._extract_response(
-                        item_id=video_id, ep='player', query=yt_age_query,
-                        ytcfg=ytcfg_age, headers=ytage_headers, fatal=False,
-                        default_client=yt_client,
-                        note='Downloading %sage-gated player API JSON' % ('android ' if force_mobile_client else '')
-                    ) or {}
+        session_index = self._extract_session_index(ytpcfg, master_ytcfg)
+        syncid = self._extract_account_syncid(ytpcfg, master_ytcfg, initial_pr)
+        sts = self._extract_signature_timestamp(video_id, player_url, master_ytcfg, fatal=False)
+        headers = self._generate_api_headers(
+            master_ytcfg, identity_token, syncid, client=self._YT_CLIENTS[client], session_index=session_index)
```

Comment:
```suggestion
        headers = self._generate_api_headers(
            ytpcfg, identity_token, syncid, client=self._YT_CLIENTS[client], session_index=session_index)
```

should be ytpcfg so we don't accidentally generate the web client's headers if something goes wrong